client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -7,17 +7,27 @@ import React from 'react';
 /**
  * Internal Dependencies
  */
+import wpcom from 'lib/wp';
```

Comment:
A redux store seems unnecessary to me, because we don't need to get or store any data. The survey data is ephemeral and I can't think of much need to persist it or use it elsewhere. We don't even need to track the `submitting` state because we don't let the survey abort the cancellation requests, even if the survey request fails.

But we also don't want to call the API directly from a component. I think we could move most of the code that is currently inside the `if ( config.isEnabled( 'upgrades/removal-survey' ) ) {` clause from line 231 to 258 into a new exported method in [client/lib/upgrades/actions/purchases.js](https://github.com/Automattic/wp-calypso/blob/master/client/lib/upgrades/actions/purchases.js) with a signature like:
```js
function submitSurvey( surveyName, siteID, surveyData ) {
```
I wouldn't think it belongs in `client/state/purchases/actions.js` because submitSurvey doesn't actually impact any redux state, unless we're ok with putting actions there based on association versus function.

@mtias would that address your concerns?